timesketch/lib/analyzers/contrib/misp_analyzer.py

```diff
@@ -0,0 +1,173 @@
+        )
+
+        if results.status_code != 200:
+            msg_error = "Error with MISP query: Status code"
+            logger.error("{} {}".format(msg_error, results.status_code))
+            return []
+        result_loc = results.json()
+        if "name" in result_loc:
+            if "Authentication failed." in result_loc["name"]:
+                logger.error("Bad API key. Please change it.")
+                return []
+        if not result_loc["response"]["Attribute"]:
+            return []
+
+        self.total_event_counter += 1
+
+        return result_loc["response"]["Attribute"]
+
+    def mark_event(self, event, result, attr):
+        """Anotate an event with data from MISP result.
+
+        Add a comment to the event.
+
+        Args:
+            event:  The OpenSearch event object that contains type of value we search
+                    for and needs to be tagged or to add an attribute.
+            result:  Dictionnary with results from MISP.
+            attr:  type of the current value.
+        """
+
+        msg = "Event that match files:   "
+        for misp_attr in result:
+            info = misp_attr["Event"]["info"]
+            id_event = misp_attr["Event"]["id"]
+            msg += f'"Event info": "{info}"'
+            msg += f"- \"url\": {self.misp_url + '/events/view/' + id_event} || "
+
+        event.add_comment(msg)
+        event.add_tags([f"MISP-{attr}"])
+        event.commit()
+
+    def query_misp(self, query, attr, timesketch_attr):
+        """Get event from timesketch, request MISP and mark event.
+
+        Args:
+            query:  Search for all events that contains 'timesketch_attr' value.
+            attr:  type of the current value.
+            timesketch_attr:  type of the current value in timesketch format.
+        """
+        events = self.event_stream(query_string=query, return_fields=[timesketch_attr])
+        create_a_view = False
+        for event in events:
+            loc = event.source.get(timesketch_attr)
+            if attr == "filename":
+                loc = ntpath.basename(loc)
+                if not loc:
+                    _, loc = ntpath.split(event.source.get(timesketch_attr))
+
+            if not loc in self.request_set:
+                result = self.get_misp_attributes(loc, attr)
+                if result:
+                    create_a_view = True
+                    self.mark_event(event, result, attr)
+                    self.result_dict[f"{attr}:{loc}"] = result
+                else:
+                    self.result_dict[f"{attr}:{loc}"] = False
+                self.request_set.add(loc)
+            elif self.result_dict[f"{attr}:{loc}"]:
+                self.mark_event(event, self.result_dict[f"{attr}:{loc}"], attr)
+
+        if create_a_view:
+            self.sketch.add_view(
+                view_name="MISP known attribute",
+                analyzer_name=self.NAME,
+                query_string=('tag:"MISP"'),
```

Comment:
```suggestion
                query_string='tag:"MISP"',
```

Reply:
Fixed in c0fd536f2fea435e66ecd4d5cf919714474952e8.